test/data/run_safe_shell_exec.py

```diff
@@ -0,0 +1,18 @@
+import sys
+import time
+
+from horovod.run.common.util import safe_shell_exec
+
+
+class FakeEvent(object):
+    def wait(self):
+        time.sleep(999)
+
+
+if __name__ == '__main__':
+    cmd = ' '.join([sys.executable] + sys.argv[1:])
+
+    # Mock out the event to avoid leaking semaphores
```

Comment:
why would this leak semaphores, why would this be problematic here, and are tests still testing properly with this mock?

Reply:
I'll add this as a comment:

Python runs a global process called Semaphore Tracker that is used across processes spawned with the multiprocessing API.  If a process spawned by multiprocessing is hard killed while holding semaphores, then they will not be properly cleaned up and will be effectively leaked.  Note that this is only a problem at test time, as in practice the process that is hard-killed will not be the child of yet another process.

Reply:
In the unit test you run the child process `run_safe_shell_exec.py` via `subprocess.Popen`. Will the Semaphore Tracker still recognize the semaphore, created in that process for the child-child process? Only `run_safe_shell_exec.py` will use the `multiprocessing` API.

Reply:
No, subprocess.Popen will spawn its own semaphore tracker.